shared/runtime/pyexec.c

```diff
@@ -143,11 +150,16 @@ static int parse_compile_execute(const void *source, mp_parse_input_kind_t input
             mp_hal_stdout_tx_strn("\x04", 1);
         }
 
-        // check for SystemExit
-        if (mp_obj_is_subclass_fast(MP_OBJ_FROM_PTR(((mp_obj_base_t *)nlr.ret_val)->type), MP_OBJ_FROM_PTR(&mp_type_SystemExit))) {
+        #if MICROPY_ENABLE_VM_ABORT
+        if (nlr.ret_val == NULL) { // abort
+            ret = pyexec_abort;
+            nlr_set_abort(NULL);
```

Comment:
I think `nlr_set_abort(NULL);` needs to be called unconditionally after the closing brace matching `if (nlr_push(&nlr) == 0) {`. This is a bit tricky though since things in the exception handler could potentially raise an exception which would cause the `nlr_set_abort(NULL);` to be skipped. On the other hand, if that does happen, we probably end up crashing on an unhandled nlr jump anyway.

Reply:
Fixed.